faker/proxy.py

```diff
@@ -235,3 +243,44 @@ def factories(self):
 
     def items(self):
         return self._factory_map.items()
+
+
+class UniqueProxy:
+    def __init__(self, proxy):
+        self._proxy = proxy
+        self._seen = {}
+        self._sentinel = object()
```

Comment:
I wanted a slightly more "fun" sentinel than a standard object, if only for a better `repr()`, but I didn't want to drag in another library just for that.